app/components/safe_area_view/safe_area_view.ios.js

```diff
@@ -103,6 +102,15 @@ export default class SafeAreaIos extends PureComponent {
         }
     };
 
+    onSafeAreaInsetsForRootViewChange = (result) => {
```

Comment:
Does the insets value change if we move between portrait or landscape?
if not, we are asking for conditions (except for `mounted`) that are fixed so we can ask for them once when we are about to add the listener and decide there for adding a simpler function which only asks if the component is mounted or not even adding the listener if there are no insets as there is nothing to change.
what do you think?